### ISSUE TITLE
How to import hammer without using `<script>` tag

### ISSUE THREAD

**Issue by bill-mybiz** [hammerjs/hammer.js#1032]
I'm using vanilla JS es6 (2015) and I'm trying to import hammer. I can't seem to find any documentation on how to use these modules instead of using the `<script>` tag. Is there any way to do this or is the `<script>` tag the only way?

**Comment by arjunkathuria** [comment]
Hammer supports Amd and requirejs styles too.
[see here:-](https://github.com/hammerjs/hammer.js/blob/master/hammer.js#L3235)

so if you're already using say require.js(which i assume you are not) in your page,  you could `var Hammer = require(['hammer.js']);`

if you want to use \ import hammer as an ES6 module, the current distributed file is transpiled to ES5, so that wont really work.

Tl;DR:- for now, go with the script tag.

**Comment by bill-mybiz** [comment]
Thanks for the confirmation @arjunkathuria. I had looked at that export definition but I was not sure about it and I couldn't find any documentation anywhere else. 

I actually can do the `require` version, but since I didn't see it on the [npm package description](https://www.npmjs.com/package/hammerjs), and being a bit frustrated by that point on attempting various permutations of module imports with spellings of hammer/Hammer/hammerjs/hammer.js/etc, I didn't explore the `require` version. After your feedback, I found the following works:

```
var Hammer = require(['hammerjs']);
```

NB that your version with the `.` does not work! :scream: Maybe a version thing :question: 

Because it's so easy to get wrong, I would recommend that you could possibly add this to your `npm` page, since using `npm` often implies avoiding the `<script>` route...But it's easy for me to say when I don't have to do the work! :seat: :football: 

But thanks for your prompt feedback without which, I might not have gotten it going! :+1: 

Btw,
> Tl;DR:- for now, go with the script tag.
This had me laughing :smile:

**Comment by arjunkathuria** [comment]
@bill-mybiz Glad you got it working 😉 

cc: @arschmitz @runspired guess this one's okay to close.

**Comment by BillyWM** [comment]
Just FYI if you (or anyone reading this) are *actually* using vanilla Javascript, then you're only using modules on Chrome for Android, and maybe Edge with a flag turned on.

Oh, and you can't get away from the script tag :)

https://www.nczonline.net/blog/2016/04/es6-module-loading-more-complicated-than-you-think/

**Comment by looeee** [comment]
Just in case anybody else finds this - actually you can use Hammer as an ES6 module. 

If it's installed via NPM, use:

```js
import * as Hammer from 'hammerjs';
window.Hammer = Hammer.default;
```

Thereafter, from the same file you can use Hammer as `window.Hammer`, and in other files you can just use `Hammer`.

**Comment by BillyWM** [comment]
@looeee: Again, you **cannot** use this feature in web browsers reliably. 

Please consult the following:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import

The only thing that has changed in the past 4 months is that Safari added support to a technology preview build.

You still **can't use this feature on Chrome, Firefox, Internet Explorer, Opera, and iOS Safari**

**Comment by looeee** [comment]
@BillyWM yes, you'll have to use some kind of build tool -  presumably all people at this point that are talking about using imports in the browser are doing so. I use Rollup.js with Babel. Alternatives are Browserify, Webpack, System.js, JSPM and probably others.

**Comment by satiewaltz** [comment]
@BillyWM No one uses imports in browsers (even with polyfills) - the most common use case where imports are used is via Webpack where the imports are compiled via Babel (Or some other module bundler).

I believe that is what most people who would need imports use it with currently.

**Comment by agnivade** [comment]
I am using Angular2, and for me

```js
import 'hammerjs';
```

worked.

**Comment by ringcrl** [comment]
> I am using Angular2, and for me
> 
> ```js
> import 'hammerjs';
> ```
> 
> worked.

Thanks!!!

**Comment by gonnavis** [comment]
> I am using Angular2, and for me
> 
> ```js
> import 'hammerjs';
> ```
> 
> worked.

Great! Thanks! It works with Vue2 too.

With eslint, I had to add `/* eslint-disable-next-line */`
```
    /* eslint-disable-next-line */
    const hammertime = new Hammer(this.$refs.webglCanvas, {});
    /* eslint-disable-next-line */
    hammertime.get('swipe').set({ direction: Hammer.DIRECTION_VERTICAL });
```
Any good solution? ( I don't want to turn off eslint no-undef )
